scripts/sklearn_to_tmva.py

```diff
@@ -0,0 +1,176 @@
+		IVar = tree.feature[node_id]
+		#print "Node", depth*" ", kind, IVar, tree.threshold[node_id], value[0]
+		return '<Node pos="{0}" depth="{1}" NCoef="0" \
+IVar="{2}" Cut="{3:.16E}" cType="1" \
+res="{4:.16E}" rms="0.0" \
+purity="{5}" nType="0">'.format(
+		kind, depth, IVar, tree.threshold[node_id], value[0] / cls.n_estimators * coef,
+		tree.impurity[node_id]
+	)
+
+
+def recurse(cls, outfile, t, coef, node_id=0, criterion="impurity", depth=0, kind="s"):   
+    outfile.write("""			"""+depth*"""	"""+node_to_str(cls, t, node_id, criterion, depth, kind, coef) + """\n""")
+    left_child = t.children_left[node_id]
+    right_child = t.children_right[node_id]
+    if depth == cls.max_depth:
+        return
+    if left_child != _tree.TREE_LEAF:
+        recurse(cls, outfile, t, coef, left_child, criterion, depth+1, kind="l")
+    #if right_child != _tree.TREE_LEAF:
+        recurse(cls, outfile, t, coef, right_child, criterion, depth+1, kind="r")
+    #if left_child == _tree.TREE_LEAF or right_child == _tree.TREE_LEAF:
+    	outfile.write("""			"""+depth*""" 	"""+"""</Node>\n""")
+    
+from sklearn.ensemble import GradientBoostingClassifier, GradientBoostingRegressor
+def gbr_to_tmva(cls, data, outfile_name, **kwargs):
+    # if not isinstance(cls, GradientBoostingRegressor):
+    #     raise ValueError("Can only export GradientBoostingRegressor")
+    # if cls.loss != "huber":
+    #     raise ValueError("TMVA assumes loss=huber")
+    #if cls.n_classes_ != 1:
+    #    raise ValueError("Currently only two-class classification supported. (regression between 0 and 1).")    
+	mva_name = kwargs.get("mva_name", "bdt")
+	coef = kwargs.get("coef", 10)
+    #feature_names = data.columns.values
+	feature_names = kwargs.get("var_names", ['none'])
+	features_min = [data[:,fn].min() for fn in range(len(feature_names))]
+	features_max = [data[:,fn].max() for fn in range(len(feature_names))]
+    
+    #Create list of variables
+	varstring = ""
+	for i in range(cls.n_features):
+		varstring += '		<Variable VarIndex="{0}" Expression="{1}" Label="{1}" Title="{1}" Unit="" Internal="{1}" Type="F" Min="{2:.16E}" Max="{3:.16E}"/>\n'.format(
+			i, feature_names[i], features_min[i], features_max[i]
+		)
+
+
+	outfile = open(outfile_name, "w")
+	outfile.write(
+"""<?xml version="1.0"?>
+<MethodSetup Method="BDT::{mva_name}">
+	<GeneralInfo>
+		<Info name="TMVA Release" value="4.2.0 [262656]"/>
+		<Info name="ROOT Release" value="5.32/00 [335872]"/>
+		<Info name="Creator" value="smoortga"/>
+		<Info name="Date" value="Sun May 31 01:31:49 2015"/>
+		<Info name="Host" value="Linux cmsdev02.cern.ch 2.6.32-431.11.2.el6.x86_64 #1 SMP Wed Mar 26 09:39:43 CET 2014 x86_64 x86_64 x86_64 GNU/Linux"/>
+		<Info name="Dir" value="/user_mnt/user/smoortga/CTag/TMVActag_v1/CTaggingCvsB/CMSSW_7_5_1_Retraining"/>
+    	<Info name="Training events" value="1692351"/>
+    	<Info name="TrainingTime" value="7.82964582e+03"/>
+    	<Info name="AnalysisType" value="Classification"/>
```

Comment:
Do these value matter? If so it would be better to set them dynamically and not hardcoded. What worries me the most are: 
   * TMVA/ROOT Release
   * Training events